### DISCUSSION TITLE
How to trigger 5G Procedures (handover, paging, idle mode etc)

### DISCUSSION THREAD

**Discussion by MShayanNazeer** [HewlettPackard/PacketRusher#132] [Q&A] [answered]
Hello everyone,

I need some help using PacketRusher with free5GC. I've been struggling to figure out how to trigger various 5G procedures in the core network, particularly the handover process. I've looked through the documentation, but it seems that a usage guide for PacketRusher isn't available yet.

Could anyone guide me on how to trigger a handover in the 5G core using PacketRusher? Any insights or resources would be greatly appreciated!

Thank you in advance for your assistance.

**Comment by linouxis9** [comment, marked as answer]
Hi @MShayanNazeer,

I'm happy to help!

To see all the possible commands, I would advise you to run the following the command: `sudo ./packetrusher multi-ue --help`.
One example is for eg. `sudo ./packetrusher multi-ue -n 1 -xnh 3000 -idl 13000 -ngh 23000`.
This command will create and register an UE, make it do a Xn Handover to another UE after 3000 ms, then a switch to CM-IDLE and then CM-CONNECTED procedure after 13000 ms, and then finally a NGAP Handover after 23000 ms.

For these procedures to work, you must be able to create two gNodeB with PacketRusher:

PacketRusher automatically creates a second gNodeB with the same configuration as the first gNodeB, but with the IP used by the second gNodeB next to the first gNodeB: if the first gNodeB has IP 10.0.0.1 in config.yml, the second gNodeB created automatically by PacketRusher will have IP 10.0.0.2 and this cannot be configured differently for now.
As such, make sure to add, for example, the second IP 10.0.0.2 to your network interface (eg. sudo ip addr add eth0 10.0.0.2).

I hope it helped, feel free to reach out if you need more help!

Thanks and best regards,
Valentin

**Reply by MShayanNazeer** [reply]
Thanks Valentin. PacketRusher is amazing tool

**Reply by linouxis9** [reply]
Thanks a ton for your kind words @MShayanNazeer!!

**Comment by alekodu** [comment]
Hi,
I'm wondering how to configure the network interfaces for the handover simulation. Does PacketRusher require that the same network device, e.g., eth0, is assigned two consecutive IP addresses for the two gNodeBs? Or can it support having different devices, e.g., eth0 & eth1, which have assigned consecutive IPs?
I would also like to understand to what extend does PacketRusher support handover? There are simple and more complex handover scenarios; e.g., intra-, inter- AMF handover, among others. Does the N2 handover of PacketRusher support inter-AMF handover (i.e., when having multiple AMF instances)?
Thanks!
